config/utils_methods.py

```diff
@@ -177,4 +177,65 @@ def remove_fields(obj):
+            remove_fields(item)
+
+##---------- ONE API - MULTIPLE API CALLS (CRUD OPERATIONS) ---------------
+# This function helps to add newly created primarykey from main table to all other models's data
+def add_key_value_to_all_ordereddicts(list_of_ordered_dict, key, value):
+    for ordered_dict in list_of_ordered_dict:
+        ordered_dict[key] = value
+
+# If the data in post method (request) have mutiple instances to be created, so this function helps to create them.
+def create_multi_instance(data_set,serializer_name):
+    data_list = []
+    for item_data in data_set:
+        serializer = serializer_name(data=item_data)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            data = serializer.data
+            data_list.append(data)
+    return data_list
+
+# If multiple instances to be updated on same model, at a single time this function helps to update all instances.
+def update_multi_instance(data_set,pk,main_model_name,current_model_name,serializer_name,main_model_field_name=None):
+    data_list = []
+    for data in data_set:
+        # main model PK Field name
+        main_model_pk_field_name = main_model_name._meta.pk.name
+        # current model PK Field name
+        current_model_field_name = current_model_name._meta.pk.name
+        # Get the value of current model's PK field
+
+        val = data.get(f'{current_model_field_name}')
+        # Arrange arguments to filter
+        if main_model_field_name is not None: # use external value if provided
+            filter_kwargs = {main_model_field_name: pk, current_model_field_name:val}
+        else:
+            filter_kwargs = {main_model_pk_field_name: pk, current_model_field_name:val}
+            
+        instance = current_model_name.objects.filter(**filter_kwargs).first()
+        serializer = serializer_name(instance, data=data, partial=False)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        data = serializer.data
+        data_list.append(data)
+    return data_list
+
+# If delete is requested on main model's instance, so that all the related data in remaing models should be deleted
+# So this function helps to delete all the data i remaing models
+def delete_multi_instance(del_value,main_model_name,current_model_name,main_model_field_name=None):
+    # main model PK Field name
+    main_model_pk_field_name = main_model_name._meta.pk.name
+
+    # Arrange arguments to filter
+    if main_model_field_name is not None: # use external value if provided
+        filter_kwargs = {main_model_field_name: del_value}
+    else:
+        filter_kwargs = {main_model_pk_field_name: del_value}
+        
+    deleted_count, _ = current_model_name.objects.filter(**filter_kwargs).delete()
+
+    if deleted_count > 0:
+        return Response(status=status.HTTP_204_NO_CONTENT)
+    else:
+        return Response({f'***error: {current_model_name} not found or already deleted.***'})            
```

Comment:
You are already returning the response from delete method. So return status.HTTP_204_NO_CONTENT is fine here based on this return the response in delete method. 
In else return HTTP_404_NOT_FOUND

Reply:
Updated